flexmeasures/data/models/time_series.py

```diff
@@ -584,6 +585,14 @@ def __init__(
         source: tb.DBBeliefSource,
         **kwargs,
     ):
+        # get a Sensor instance attached to the database session (input sensor is detached)
+        # check out Issue #683 for more details
+        inspection_obj = inspect(sensor, raiseerr=False)
+        if inspection_obj:
+            # fetch Sensor only when it is detached
+            if inspection_obj.detached:
```

Comment:
If statements in Python are evaluated from left to right and break early, so you could simplify this to:
```
if inspection_obj and inspection_obj.detached:
```

Reply:
True, much better!